### ISSUE TITLE
OCaml 5.2 support

### ISSUE THREAD

**Issue by kit-ty-kate** [ocaml-batteries-team/batteries-included#1127]
The current version of batteries is not compatible with OCaml 5.2 (currently released as 5.2.0\~alpha1)

**Comment by UnixJunkie** [comment]
Do you have a list of errors?

**Comment by vbgl** [comment]
~~~
File "src/batPrintexc.mli", lines 236-241, characters 0-1:
Error: This variant or record definition does not match that of type
         Printexc.location
       5. An extra field, end_line, is provided in the original definition.
       6. An extra field, end_col, is provided in the original definition.
~~~

~~~
File "src/batGc.pp.ml", line 1:
Error: The implementation src/batGc.pp.ml
       does not match the interface src/batGc.pp.ml:  ... In module Memprof:
       Values do not match:
         val start :
           sampling_rate:float ->
           ?callstack_size:int -> ('minor, 'major) tracker -> t
       is not included in
         val start :
           sampling_rate:float ->
           ?callstack_size:int -> ('minor, 'major) tracker -> unit
       The type
         sampling_rate:float -> ?callstack_size:int -> ('a, 'b) tracker -> t
       is not compatible with the type
         sampling_rate:float ->
         ?callstack_size:int -> ('a, 'b) tracker -> unit
       Type t = Gc.Memprof.t is not compatible with type unit
       File "src/batGc.mli", lines 475-479, characters 4-10:
         Expected declaration
       File "gc.mli", lines 498-502, characters 4-7: Actual declaration
~~~

~~~
File "src/batBigarray.mli", lines 111-124, characters 0-50:
Error: This variant or record definition does not match that of type
         ('a, 'b) Bigarray.kind
       An extra constructor, Float16, is provided in the original definition.
~~~

~~~
File "src/batGc.pp.ml", line 1:
Error: The implementation src/batGc.pp.ml
       does not match the interface src/batGc.pp.ml:  ... In module Memprof:
       Values do not match:
         val start :
           sampling_rate:float ->
           ?callstack_size:int -> ('minor, 'major) tracker -> t
       is not included in
         val start :
           sampling_rate:float ->
           ?callstack_size:int -> ('minor, 'major) tracker -> unit
       The type
         sampling_rate:float -> ?callstack_size:int -> ('a, 'b) tracker -> t
       is not compatible with the type
         sampling_rate:float ->
         ?callstack_size:int -> ('a, 'b) tracker -> unit
       Type t = Gc.Memprof.t is not compatible with type unit
       File "src/batGc.mli", lines 475-479, characters 4-10:
         Expected declaration
       File "gc.mli", lines 498-502, characters 4-7: Actual declaration
~~~

**Comment by UnixJunkie** [comment]
I don't think I will have time before two weeks for that...

**Comment by kit-ty-kate** [comment]
I'll try to open a PR this week then

**Comment by kit-ty-kate** [comment]
Done https://github.com/ocaml-batteries-team/batteries-included/pull/1128

**Comment by UnixJunkie** [comment]
This is included in tag v3.8.0.
